docs/serving/autoscaling/autoscaling-concepts.md

```diff
@@ -0,0 +1,135 @@
+
+* Not part of the Knative Serving core, and must be enabled after [Knative Serving installation](../../install/any-kubernetes-cluster.md#optional-serving-extensions).
+* Does not support scale to zero functionality.
+* Supports CPU-based autoscaling.
+
+### Configuring the Autoscaler implementation
+
+The type of Autoscaler implementation (KPA or HPA) can be configured by using the `class` annotation.
+
+* **Global settings key:** `pod-autoscaler-class`
+* **Per-revision annotation key:** `autoscaling.knative.dev/class`
+* **Possible values:** `"kpa.autoscaling.knative.dev"` or `"hpa.autoscaling.knative.dev"`
+* **Default:** `"kpa.autoscaling.knative.dev"`
+
+**Example:**
+{{< tabs name="class" default="Per Revision" >}}
+{{% tab name="Per Revision" %}}
+```yaml
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: helloworld-go
+  namespace: default
+spec:
+  template:
+    metadata:
+      annotations:
+        autoscaling.knative.dev/class: "kpa.autoscaling.knative.dev"
+    spec:
+      containers:
+        - image: gcr.io/knative-samples/helloworld-go
+```
+{{< /tab >}}
+{{% tab name="Global (ConfigMap)" %}}
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+ name: config-autoscaler
+ namespace: knative-serving
+data:
+ pod-autoscaler-class: "kpa.autoscaling.knative.dev"
+```
+{{< /tab >}}
+{{% tab name="Global (Operator)" %}}
+```yaml
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+spec:
+  config:
+    autoscaler:
+      pod-autoscaler-class: "kpa.autoscaling.knative.dev"
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Global versus per-revision settings
+
+Configuring for autoscaling in Knative can be set using either global or per-revision settings.
+
+1. If no per-revision autoscaling settings are specified, the global settings will be used.
+1. If per-revision settings are specified, these will override the global settings when both types of settings exist.
+
+### Global settings
+
+Global settings for autoscaling are configured using the `config-autoscaler` ConfigMap. If you installed Knative Serving using the Operator, you can set global configuration settings in the `spec.config.autoscaler` ConfigMap, located in the `KnativeServing` custom resource (CR).
+
+#### Example of the default Kubernetes ConfigMap
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+ name: config-autoscaler
+ namespace: knative-serving
+data:
+ container-concurrency-target-default: "100"
+ container-concurrency-target-percentage: "0.7"
+ enable-scale-to-zero: "true"
+ max-scale-up-rate: "1000"
+ max-scale-down-rate: "2"
+ panic-window-percentage: "10"
+ panic-threshold-percentage: "200"
+ scale-to-zero-grace-period: "30s"
+ stable-window: "60s"
+ tick-interval: "2s"
+ target-burst-capacity: "200"
+ requests-per-second-target-default: "200"
+```
+
+### Per-revision settings
+
+Per-revision settings for autoscaling are configured by adding _annotations_ to a revision.
```

Comment:
Are the annotation keys the same ones as are present in the ConfigMap?

Reply:
Nope, they're different afaik, it's explained in the section for each key.